cdap-proto/src/test/java/co/cask/cdap/proto/id/EntityIdTest.java

```diff
@@ -188,6 +188,18 @@ private void doTestToFromOldId(EntityId id) {
   }
 
   @Test
+  public void testDatasetName() {
+    // TODO: CDAP-5730: Even though dots are allowed, if we use it here, id creation will fail.
```

Comment:
yeah, I meant that we shouldn't start using these EntityId classes everywhere (especially places where we serialize them) before we fix these shortcomings